src/pages/index.js

```diff
@@ -65,35 +86,128 @@ profileEditButton.addEventListener("click", () => {
   editPopupForm.open();
 });
 
+//Avatar Profile change
+avatarEditButton.addEventListener("click", () => {
+  avatarFormValidator.toggleButtonState();
+  avatarProfilePopupForm.open();
+});
+
 /*Functions*/
 /*Cards*/
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleCardImageClick);
-  section.addItem(card.getCardElement());
+  const card = new Card({
+    data: cardData,
+    templateSelector: "#card-template",
+    handleCardImageClick,
+    myUserId: userId,
+    handleDeleteCard,
+    handleLikeButton: (cardId) => {
+      if (card.isLiked()) {
+        api
+          .removeLike(cardId)
+          .then((data) => {
+            card.updateLikes(data.likes);
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      } else {
+        api
+          .addLike(cardId)
+          .then((data) => {
+            card.updateLikes(data.likes);
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      }
+    },
+  });
+  return card.getCardElement();
+}
+let cardList;
+api
+  .getInitialCards()
+  .then((initialCards) => {
+    cardList = new Section(
+      {
+        items: initialCards,
+        renderer: (data) => {
+          const newCard = renderCard(data);
+          cardList.addItem(newCard);
+        },
+      },
+      cardsWrap
+    );
+    cardList.renderItems();
+  })
+  .catch((err) => console.log(err));
+
+let userId;
+api
+  .getUserInfo()
+  .then((formData) => {
+    userId = formData._id;
+    userInfo.setProfileAvatar(formData.avatar);
+    userInfo.setUserInfo(formData.name, formData.about);
+  })
+  .catch((err) => console.log(err));
+
+function handleEditProfileSubmit({ nameInfo, jobInfo }) {
+  api
+    .changeUserInfo(nameInfo, jobInfo)
+    .then(() => {
+      userInfo.setUserInfo(nameInfo, jobInfo);
+      editPopupForm.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-  },
-  cardsWrap
-);
-
-section.renderItems();
-
-// Handlers/
-
-function handleEditProfileSubmit(formData) {
-  const { nameInfo, jobInfo } = formData;
-  userInfo.setUserInfo(nameInfo, jobInfo);
-  editPopupForm.close();
+function handleAddCardFormSubmit({ name, link }) {
+  api
+    .addCard(name, link)
+    .then((cardElement) => {
+      const newCard = renderCard(cardElement);
+      cardList.prependItem(newCard);
+      addCardPopupForm.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+//Delete Card
+function handleDeleteCard(cardId) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.renderLoading(true);
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        this.deleteConfirmClick();
```

Comment:
`this` should **not** be used outside classes in functions in `index.js` . This context is easy to lose. You need to find other ways of passing data.


Something like that

```js

function handleDeleteCard(cardId, cardInstance) {
  deleteCardPopup.open();
  deleteCardPopup.setSubmitAction(() => {
    deleteCardPopup.renderLoading(true);
    api
      .deleteCard(cardId)
      .then(() => {
        cardInstance.deleteConfirmClick();
```